New nexion project added

Lefgk opened the pull request:
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
Nexion

##### Twitter Link:
https://x.com/nexionpulse

##### List of audit links if any:


##### Website Link:
https://nexionpulse.com/

##### Logo (High resolution, will be shown with rounded borders):


##### Current TVL:


##### Treasury Addresses (if the protocol has treasury)


##### Chain:


##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)


##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)


##### Short Description (to be shown on DefiLlama):
NEXION — Built on PulseChain, powered by NEON
NEXION aims to become a widely used dApp ecosystem that offers users a variety of ways to participate and earn rewards!

##### Token address and ticker if any:
0xF2Da3942616880E52e841E5C504B5A9Fba23FFF0
NEON

##### Category (full list at https://defillama.com/categories) *Please choose only one:
[Yield](https://defillama.com/protocols/yield)

##### Oracle Provider(s): Specify the oracle(s) used (e.g., Chainlink, Band, API3, TWAP, etc.):
##### Implementation Details: Briefly describe how the oracle is integrated into your project:
##### Documentation/Proof: Provide links to documentation or any other resources that verify the oracle's usage:

##### forkedFrom (Does your project originate from another project):


##### methodology (what is being counted as tvl, how is tvl being calculated):
We add total value of Buy and Burn contract and the total assets in Holovault farm.

##### Github org/user (Optional, if your code is open source, we can track activity):

llamabutler commented:
Error while running adapter at projects/nexion: 
        
 
  
    
    
    ReferenceError: ADDRESSES is not defined
    Truncated error stack:
    
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/nexion/index.js:22:10)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:127:12
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:259:3)
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'

llamabutler commented:
The adapter at projects/nexion exports TVL: 
        
 
 
    pulse                     15.85 k
    pulse-staking             0
    staking                   0
    
    total                    15.84 k

llamabutler commented:
The adapter at projects/nexion exports TVL: 
        
 
 
    pulse                     15.85 k
    pulse-staking             0
    staking                   0
    
    total                    15.84 k

llamabutler commented:
The adapter at projects/nexion exports TVL: 
        
 
 
    pulse                     15.85 k
    pulse-staking             0
    staking                   0
    
    total                    15.84 k

llamabutler commented:
The adapter at projects/nexion exports TVL: 
        
 
 
    pulse                     15.85 k
    pulse-staking             0
    pulse-pool2               0
    staking                   0
    pool2                     0
    
    total                    15.84 k

g1nt0ki commented on projects/nexion/index.js:
```diff
@@ -0,0 +1,64 @@
+
+  const plsin = await api.call({
+    target: contracts.NEONFarm,
+    abi: farmAbi.valueofthis,
```
hmm, what does this call return?

Lefgk replied:
A portion of user deposits are sent to this farm contract. valueofthis() tracks the plsvalue received.

 function valueofthis() external view returns (uint256 plsvalue) {
        address[] memory path = new address[](2);
        path[0] = address(PLS);
        path[1] = address(DAI);

        uint256 price = ISwapRouter(PULSEX_V2).getAmountsOut(10 ** 18, path)[1];
        plsvalue = (plsin * price) / 10 ** 18;
    }
    
    This is then used to buy DAI-PLS LP and the LP is burned. It stays there forever and rewards users. We rehypothecate on this farm.

g1nt0ki replied:
yeah, no. If the LP is burnt, it is not tvl, we can count what users can withdraw as tvl

g1nt0ki replied:
sorry, yeah. you need to leave it out

Lefgk replied:
ok what about native token and LP? 
We also have a staking where users stake NEON and earn NEON and INC token. 
we also have around 9k $ in LP (NEON/PLS, NEON/DAI etc..all NEON pairs) Can i add these as tvl?

g1nt0ki replied:
we can count staked NEON under staking
We can count staked NEON/PLS, etc LPS under pool2


But we can count these only if the user can withdraw their deposits, else no.

g1nt0ki commented on projects/nexion/index.js:
```diff
@@ -0,0 +1,64 @@
+async function tvl(api) {
+  return sumTokens2({
+    api,
+    owner: contracts.NEONBuynBurn,
```
sorry, we can treat only the user deposits as tvl

Lefgk replied:
A portion of user deposits go to buyNBurn contract. So these funds are from user deposits. Or am i missing something?

g1nt0ki replied:
can you explain the mechanism, thought this comes from fees, that is usually the case when protocols use the term buyback and burn

g1nt0ki commented on projects/nexion/index.js:
```diff
@@ -0,0 +1,64 @@
+  sdk.util.sumSingleBalance(balances, contracts.NEONFarm, plsin2);
+  return balances;
+}
+const lps = ["0xE56043671df55dE5CDf8459710433C10324DE0aE"]
```
since this is DAI-PLS LP and has nothing to do with your protocol, you can count it as tvl instead of pool2 if you want

Lefgk replied:
ok

g1nt0ki reviewed (changes requested):
.

llamabutler commented:
Error while running adapter at projects/nexion: 
        
 
  
    
    
    TypeError: i.toLowerCase is not a function
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/tokenMapping.js:143:50
        at Array.forEach (<anonymous>)
        at getUniqueAddresses (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/tokenMapping.js:143:13)
        at sumTokens2 (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/unwrapLPs.js:795:12)
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/nexion/index.js:31:10)
        at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:29)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:173:15
        at Array.map (<anonymous>)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:161:29
        at Array.map (<anonymous>)
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'

llamabutler commented:
Error while running adapter at projects/nexion: 
        
 
  
    
    
    TypeError: i.toLowerCase is not a function
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/tokenMapping.js:143:50
        at Array.forEach (<anonymous>)
        at getUniqueAddresses (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/tokenMapping.js:143:13)
        at sumTokens2 (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/unwrapLPs.js:795:12)
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/nexion/index.js:31:10)
        at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:29)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:173:15
        at Array.map (<anonymous>)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:161:29
        at Array.map (<anonymous>)
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'

llamabutler commented:
Error while running adapter at projects/nexion: 
        
 
  
    
    
    TypeError: i.toLowerCase is not a function
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/tokenMapping.js:144:50
        at Array.forEach (<anonymous>)
        at getUniqueAddresses (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/tokenMapping.js:144:13)
        at sumTokens2 (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/unwrapLPs.js:795:12)
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/nexion/index.js:31:10)
        at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:29)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:173:15
        at Array.map (<anonymous>)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:161:29
        at Array.map (<anonymous>)
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'

llamabutler commented:
The adapter at projects/nexion exports TVL: 
        
 
 
    pulse                     15.73 k
    pulse-staking             0
    staking                   0
    
    total                    15.73 k

llamabutler commented:
Error while running adapter at projects/nexion: 
        
 
  
    
    
    [object Object]

llamabutler commented:
The adapter at projects/nexion exports TVL: 
        
 
 
    pulse                     15.74 k
    
    total                    15.74 k

llamabutler commented:
The adapter at projects/nexion exports TVL: 
        
 
 
    pulse                     15.69 k
    
    total                    15.69 k

llamabutler commented:
The adapter at projects/nexion exports TVL: 
        
 
 
    pulse                     17.05 k
    
    total                    17.05 k

Lefgk commented:
> .

I added back DAI-PLS LP as tvl

g1nt0ki commented:
Havent approved this PR because, you are still counting the tvl in buyback and burn wallet

"
we can count staked NEON under staking
We can count staked NEON/PLS, etc LPS under pool2


But we can count these only if the user can withdraw their deposits, else no."

Lefgk commented:
> Havent approved this PR because, you are still counting the tvl in buyback and burn wallet
> 
> " we can count staked NEON under staking We can count staked NEON/PLS, etc LPS under pool2
> 
> But we can count these only if the user can withdraw their deposits, else no."

Hello,

i added staking for neon.  Users can withdraw this anytime without fees.
I removed tvl buyback + burn wallet

llamabutler commented:
Error while running adapter at projects/nexion: 
        
 
  
    
    
    Error: Cannot find module './helper/staking'
    Require stack:
    - /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/nexion/index.js
    - /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js
    Truncated error stack:
    
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/nexion/index.js:6:21)
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'

llamabutler commented:
The adapter at projects/nexion exports TVL: 
        
 
 
    pulse                     59
    pulse-staking             0
    staking                   0
    
    total                    59.454434653734936

llamabutler commented:
The adapter at projects/nexion exports TVL: 
        
 
 
    pulse                     59
    pulse-staking             0
    staking                   0
    
    total                    59.454434653734936

llamabutler commented:
The adapter at projects/nexion exports TVL: 
        
 
 
    pulse-staking             222.22 k
    staking                   222.22 k
    pulse                     135
    
    total                    135.26253176291314
